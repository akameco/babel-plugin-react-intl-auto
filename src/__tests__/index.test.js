// @flow
import { resolve } from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '../'

const rootPath = resolve(__dirname, '../__fixtures__')
const filename = resolve(rootPath, 'messages.js')

const basicTest = {
  title: 'basic',
  code: `
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})
      `,
}

const multiExportTest = {
  title: 'multi export',
  code: `
import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello extra',
  world: 'hello world extra',
})

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})
`,
}

const tests = [
  basicTest,
  {
    title: 'with include value',
    code: `
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})
      `,
  },
  {
    title: 'string literal',
    code: `
import { defineMessages } from 'react-intl'

defineMessages({
  'hello': 'hello',
  'world': 'hello world',
})
      `,
  },
  {
    title: 'Object',
    code: `
import { defineMessages } from 'react-intl'

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id',
  },
  world: {
    defaultMessage: 'world',
  },
  headerTitle: {
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.',
  },
})
      `,
  },
  {
    title: 'import as',
    code: `
import { defineMessages as m } from 'react-intl'

m({
  hello: 'hello',
  world: 'hello world',
})

`,
  },
  {
    title: 'with other func',
    code: `
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

hello({
  id: 'hoge',
})
    `,
  },
  multiExportTest,
]

const defaultOpts = {
  plugin,
  snapshot: true,
  babelOptions: { filename },
}

pluginTester({
  ...defaultOpts,
  tests,
})

pluginTester({
  ...defaultOpts,
  title: 'removePrefix = "src"',
  tests,
  pluginOptions: { removePrefix: 'src' },
})

pluginTester({
  ...defaultOpts,
  title: 'removePrefix = "src/" -- with slash',
  tests,
  pluginOptions: { removePrefix: 'src/' },
})

pluginTester({
  ...defaultOpts,
  title: 'filebase = true',
  tests,
  pluginOptions: { filebase: true },
})

pluginTester({
  ...defaultOpts,
  title: 'includeExportName = true',
  tests: [basicTest, multiExportTest],
  pluginOptions: { includeExportName: true },
})

pluginTester({
  ...defaultOpts,
  title: 'includeExportName = all',
  tests: [basicTest, multiExportTest],
  pluginOptions: { includeExportName: 'all' },
})
