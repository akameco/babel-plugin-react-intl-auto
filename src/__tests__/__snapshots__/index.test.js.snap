// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Object 1`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id',
  },
  world: {
    defaultMessage: 'world',
  },
  headerTitle: {
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.',
  },
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': 'src.__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': 'src.__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});
"
`;

exports[`Object 2`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id',
  },
  world: {
    defaultMessage: 'world',
  },
  headerTitle: {
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.',
  },
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': '__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': '__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});
"
`;

exports[`Object 3`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id',
  },
  world: {
    defaultMessage: 'world',
  },
  headerTitle: {
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.',
  },
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': '__fixtures__.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': '__fixtures__.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});
"
`;

exports[`Object 4`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id',
  },
  world: {
    defaultMessage: 'world',
  },
  headerTitle: {
    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.',
  },
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  new: {
    id: 'this is id',
    defaultMessage: 'id'
  },
  world: {
    'id': 'src.__fixtures__.messages.world',

    defaultMessage: 'world'
  },
  headerTitle: {
    'id': 'src.__fixtures__.messages.headerTitle',

    defaultMessage: 'Welcome to dashboard {name}!',
    description: 'Message to greet the user.'
  }
});
"
`;

exports[`basic 1`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`basic 2`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`basic 3`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`basic 4`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  hello: {
    'id': 'src.__fixtures__.messages.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.messages.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`basic 5`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`basic 6`] = `
"
import { defineMessages } from 'react-intl'

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export default defineMessages({
  hello: {
    'id': 'src.__fixtures__.default.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.default.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`import as 1`] = `
"
import { defineMessages as m } from 'react-intl'

m({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages as m } from 'react-intl';

m({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`import as 2`] = `
"
import { defineMessages as m } from 'react-intl'

m({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages as m } from 'react-intl';

m({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`import as 3`] = `
"
import { defineMessages as m } from 'react-intl'

m({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages as m } from 'react-intl';

m({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`import as 4`] = `
"
import { defineMessages as m } from 'react-intl'

m({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages as m } from 'react-intl';

m({
  hello: {
    'id': 'src.__fixtures__.messages.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.messages.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`multi export 1`] = `
"
import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello extra',
  world: 'hello world extra',
})

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export const extra = defineMessages({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world extra'
  }
});

export default defineMessages({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`multi export 2`] = `
"
import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello extra',
  world: 'hello world extra',
})

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export const extra = defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world extra'
  }
});

export default defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`multi export 3`] = `
"
import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello extra',
  world: 'hello world extra',
})

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export const extra = defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world extra'
  }
});

export default defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`multi export 4`] = `
"
import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello extra',
  world: 'hello world extra',
})

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export const extra = defineMessages({
  hello: {
    'id': 'src.__fixtures__.messages.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.messages.world',
    'defaultMessage': 'hello world extra'
  }
});

export default defineMessages({
  hello: {
    'id': 'src.__fixtures__.messages.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.messages.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`multi export 5`] = `
"
import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello extra',
  world: 'hello world extra',
})

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export const extra = defineMessages({
  hello: {
    'id': 'src.__fixtures__.extra.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.extra.world',
    'defaultMessage': 'hello world extra'
  }
});

export default defineMessages({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`multi export 6`] = `
"
import { defineMessages } from 'react-intl'

export const extra = defineMessages({
  hello: 'hello extra',
  world: 'hello world extra',
})

export default defineMessages({
  hello: 'hello',
  world: 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

export const extra = defineMessages({
  hello: {
    'id': 'src.__fixtures__.extra.hello',
    'defaultMessage': 'hello extra'
  },
  world: {
    'id': 'src.__fixtures__.extra.world',
    'defaultMessage': 'hello world extra'
  }
});

export default defineMessages({
  hello: {
    'id': 'src.__fixtures__.default.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.default.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`string literal 1`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  'hello': 'hello',
  'world': 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  'hello': {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  'world': {
    'id': 'src.__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`string literal 2`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  'hello': 'hello',
  'world': 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  'hello': {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  'world': {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`string literal 3`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  'hello': 'hello',
  'world': 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  'hello': {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  'world': {
    'id': '__fixtures__.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`string literal 4`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  'hello': 'hello',
  'world': 'hello world',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  'hello': {
    'id': 'src.__fixtures__.messages.hello',
    'defaultMessage': 'hello'
  },
  'world': {
    'id': 'src.__fixtures__.messages.world',
    'defaultMessage': 'hello world'
  }
});
"
`;

exports[`with include value 1`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});
"
`;

exports[`with include value 2`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});
"
`;

exports[`with include value 3`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});
"
`;

exports[`with include value 4`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  hello: {
    'id': 'src.__fixtures__.messages.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.messages.world',
    'defaultMessage': \`hello world \${1}\`
  }
});
"
`;

exports[`with other func 1`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

hello({
  id: 'hoge',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  hello: {
    'id': 'src.__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});
"
`;

exports[`with other func 2`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

hello({
  id: 'hoge',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});
"
`;

exports[`with other func 3`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

hello({
  id: 'hoge',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  hello: {
    'id': '__fixtures__.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': '__fixtures__.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});
"
`;

exports[`with other func 4`] = `
"
import { defineMessages } from 'react-intl'

defineMessages({
  hello: 'hello',
  world: \`hello world \${1}\`,
})

hello({
  id: 'hoge',
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { defineMessages } from 'react-intl';

defineMessages({
  hello: {
    'id': 'src.__fixtures__.messages.hello',
    'defaultMessage': 'hello'
  },
  world: {
    'id': 'src.__fixtures__.messages.world',
    'defaultMessage': \`hello world \${1}\`
  }
});

hello({
  id: 'hoge'
});
"
`;
